Added documentation for persist verified method and Added response data (#110)
modularcloud/explorer: apps/verify/README.md

Description:

Verifies the provided smart contract code against the deployed contract address. This endpoint is used to ensure that the deployed contract matches the submitted source code. 

Request Body :

```json
{
  "contractAddress": "string",

  "files": "Object",

  "chain": "string"
}
```
Example Data:
``` {
  "contractAddress": "0x329d0c10A7d86f9d80d520B78990e46224720131",
  "chain": "88002",
  "files": {

>     "HelloWorld_metadata.json": "{\n\t\"compiler\": {\n\t\t\"version\": \"0.8.18+commit.87f61d96\"\n\t},\n\t\"language\": \"Solidity\",\n\t\"output\": {\n\t\t\"abi\": [\n\t\t\t{\n\t\t\t\t\"inputs\": [],\n\t\t\t\t\"stateMutability\": \"nonpayable\",\n\t\t\t\t\"type\": \"constructor\"\n\t\t\t},\n\t\t\t{\n\t\t\t\t\"inputs\": [],\n\t\t\t\t\"name\": \"speak\",\n\t\t\t\t\"outputs\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"internalType\": \"string\",\n\t\t\t\t\t\t\"name\": \"\",\n\t\t\t\t\t\t\"type\": \"string\"\n\t\t\t\t\t}\n\t\t\t\t],\n\t\t\t\t\"stateMutability\": \"view\",\n\t\t\t\t\"type\": \"function\"\n\t\t\t}\n\t\t],\n\t\t\"devdoc\": {\n\t\t\t\"kind\": \"dev\",\n\t\t\t\"methods\": {},\n\t\t\t\"version\": 1\n\t\t},\n\t\t\"userdoc\": {\n\t\t\t\"kind\": \"user\",\n\t\t\t\"methods\": {},\n\t\t\t\"version\": 1\n\t\t}\n\t},\n\t\"settings\": {\n\t\t\"compilationTarget\": {\n\t\t\t\"HelloWorld.sol\": \"HelloWorld\"\n\t\t},\n\t\t\"evmVersion\": \"paris\",\n\t\t\"libraries\": {},\n\t\t\"metadata\": {\n\t\t\t\"bytecodeHash\": \"ipfs\"\n\t\t},\n\t\t\"optimizer\": {\n\t\t\t\"enabled\": false,\n\t\t\t\"runs\": 200\n\t\t},\n\t\t\"remappings\": []\n\t},\n\t\"sources\": {\n\t\t\"HelloWorld.sol\": {\n\t\t\t\"keccak256\": \"0x8f54065335b9a9019f30a20b9e2cb182aeec5d823ded5c67e71c22f42422b892\",\n\t\t\t\"license\": \"MIT\",\n\t\t\t\"urls\": [\n\t\t\t\t\"bzz-raw://21b5e98caff1f4831c23977aa77c9063fe735d843cb722ebda428715f586dcf2\",\n\t\t\t\t\"dweb:/ipfs/QmSk3zRdU2fxWEaw7em9PJ6LJuupxWgM1snGwqbNZgBB2W\"\n\t\t\t]\n\t\t}\n\t},\n\t\"version\": 1\n}",

Reviewer: ```suggestion
    "HelloWorld_metadata.json": "{
&nbsp;&nbsp;&nbsp;&nbsp;"compiler": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"version": "0.8.18+commit.87f61d96"
&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;"language": "Solidity",
&nbsp;&nbsp;&nbsp;&nbsp;"output": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"abi": [
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"inputs": [],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"stateMutability": "nonpayable",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"type": "constructor"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"inputs": [],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"name": "speak",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"outputs": [
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;{
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"internalType": "string",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"name": "",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"type": "string"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"stateMutability": "view",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"type": "function"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"devdoc": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"kind": "dev",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"methods": {},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"version": 1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"userdoc": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"kind": "user",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"methods": {},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"version": 1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;"settings": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"compilationTarget": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"HelloWorld.sol": "HelloWorld"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"evmVersion": "paris",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"libraries": {},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"metadata": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"bytecodeHash": "ipfs"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"optimizer": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"enabled": false,
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"runs": 200
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"remappings": []
&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;"sources": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"HelloWorld.sol": {
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"keccak256": "0x8f54065335b9a9019f30a20b9e2cb182aeec5d823ded5c67e71c22f42422b892",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"license": "MIT",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"urls": [
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"bzz-raw://21b5e98caff1f4831c23977aa77c9063fe735d843cb722ebda428715f586dcf2",
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;"dweb:/ipfs/QmSk3zRdU2fxWEaw7em9PJ6LJuupxWgM1snGwqbNZgBB2W"
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;}
&nbsp;&nbsp;&nbsp;&nbsp;},
&nbsp;&nbsp;&nbsp;&nbsp;"version": 1
}"
```

>     "HelloWorld.sol": "// SPDX-License-Identifier: MIT\npragma solidity ^0.8.17;\n\ncontract HelloWorld {\n\n    string saySomething;\n\n    constructor() {\n        saySomething = \"Hello World!\";\n    }\n\n    function speak() public view returns(string memory) {\n        return saySomething;\n    }\n}"

Reviewer: ```suggestion
    "HelloWorld.sol": "// SPDX-License-Identifier: MIT
    pragma solidity ^0.8.17;
    
    contract HelloWorld {
    
            string saySomething;
            
                    constructor() {
                            saySomething = "Hello World!";
                                }
                                
                                        function speak() public view returns(string memory) {
                                                return saySomething;
                                                    }
                                                    }"
```

  }
} 
```

The body of the request must be a JSON object with the following properties:

- address: The smart contract address that needs to be verified.

- files: The contract source files inside a string.

- chain: The ID of the blockchain where the contract has been deployed.

Responses:

- 200 OK: The contract was verified successfully. The JSON response body will contain information about the contract address and if the contract was verified fully or partially.
	-	eg: 
	```
	 { "result": [ { "address": "0x123f681646d4a755815f9cb19e1acc8565a0c2ac", "chainId": "1", "status": "perfect", "libraryMap": { "lib1": "0x3f681646d4a755815f9cb19e1acc8565a0c2ac", "lib2": "0x4f681646d4a755815f9cb19e1acc8565a0c2ac" } } ] }  

> - 400 Bad Request: It can be returned due to errors mentioned below 

Reviewer: ```suggestion
- 400 Bad Request: It cannot be returned due to the errors mentioned below 
```

	- Unexpected token in JSON:
		eg.: 
		```
		 {  "error":  "Unexpected token ' in JSON at position 107"  }
		 ```
	- Metadata file not found.
		eg.:
		```

> 		{  "error":  "Metadata file not found. Did you include \"metadata.json\"?"  }

Reviewer: ```suggestion
		{  "error":  "Metadata file not found. Did you include "metadata.json"?"  }
```

Author: This error is  sent by souricfy

		 ```
	 - Validation Error  address:
	 eg: 	 
		 ```
		{"message":  "Validation Error: address", "errors":  [{"field":"address" "message":  "Invalid addresses: Contract Address"}]}
	
- 404 Not found:  
	- Couldn't extract files from the request.
		eg: 
		```
		{  "error":  "Couldn't extract files from the request. Please make sure you have added files"  }
		 ```
- 500 Internal Server Error:  

> 	-	All the other errors including compilation mismatch and resource missing 

Reviewer: ```suggestion
	-	All the other errors including compilation mismatch and resource(s) missing 
```

**/api/contract-verification/persist-verified**

Request method:

POST

Description: 
This endpoint is used to persist the verified contracts in the database. 

Request Body :
 ``` 
 {"verificationStatus": "string",
  "contractAddress": "string",
  "chainID": "string",
  "isVerified": "boolean",
  "uploadedUrl": "string",}
 ```
 The body of the request must be a JSON object with the following properties:

> - verificationStatus: Indicates  whether the contract is fully verified or partially verified 

Reviewer: ```suggestion
- verificationStatus: Indicates whether the contract is fully verified or partially verified 
```

> - contractAddress:The smart contract address Which has been verified.

Reviewer: ```suggestion
- contractAddress: The smart contract address Which has been verified.
```

> - chainId:The ID of the blockchain where the contract has been deployed.

Reviewer: ```suggestion
- chainId: The ID of the blockchain where the contract has been deployed.
```
